File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/injector/component/AndroidBindingModule.java
package technology.mainthread.apps.gatekeeper.injector.component;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import technology.mainthread.apps.gatekeeper.service.GatekeeperStateService;
import technology.mainthread.apps.gatekeeper.service.MessagingInstanceIdService;
import technology.mainthread.apps.gatekeeper.service.MessagingService;
import technology.mainthread.apps.gatekeeper.service.MobileWearListenerService;
import technology.mainthread.apps.gatekeeper.service.RefreshFCMSubscriptionsService;
import technology.mainthread.apps.gatekeeper.view.activity.AuthActivity;
import technology.mainthread.apps.gatekeeper.view.activity.MainActivity;
import technology.mainthread.apps.gatekeeper.view.activity.SettingsActivity;
import technology.mainthread.apps.gatekeeper.view.fragment.ImagesFragment;
import technology.mainthread.apps.gatekeeper.view.fragment.LogsFragment;
import technology.mainthread.apps.gatekeeper.view.fragment.SettingsFragment;
import technology.mainthread.apps.gatekeeper.view.fragment.UnlockFragment;

@Module
abstract class AndroidBindingModule {

    // Activity
    @ContributesAndroidInjector
    abstract MainActivity mainActivity();

    @ContributesAndroidInjector
    abstract AuthActivity authActivity();

    @ContributesAndroidInjector
    abstract SettingsActivity settingsActivity();

    // Fragment
    @ContributesAndroidInjector
    abstract ImagesFragment imagesFragment();

    @ContributesAndroidInjector
    abstract LogsFragment logsFragment();

    @ContributesAndroidInjector
    abstract SettingsFragment settingsFragment();

    @ContributesAndroidInjector
    abstract UnlockFragment unlockFragment();

    // Service
    @ContributesAndroidInjector
    abstract GatekeeperStateService gatekeeperStateService();

    @ContributesAndroidInjector
    abstract MessagingInstanceIdService messagingInstanceIdService();

    @ContributesAndroidInjector
    abstract MessagingService messagingService();

    @ContributesAndroidInjector
    abstract MobileWearListenerService mobileWearListenerService();

    @ContributesAndroidInjector
    abstract RefreshFCMSubscriptionsService refreshFCMSubscriptionsService();
}

File: mobile/src/main/java/technology/mainthread/apps/gatekeeper/injector/component/AndroidBindingModule.kt
package technology.mainthread.apps.gatekeeper.injector.component

import dagger.Module
import dagger.android.ContributesAndroidInjector
import technology.mainthread.apps.gatekeeper.service.*
import technology.mainthread.apps.gatekeeper.view.activity.AuthActivity
import technology.mainthread.apps.gatekeeper.view.activity.MainActivity
import technology.mainthread.apps.gatekeeper.view.activity.SettingsActivity
import technology.mainthread.apps.gatekeeper.view.fragment.ImagesFragment
import technology.mainthread.apps.gatekeeper.view.fragment.LogsFragment
import technology.mainthread.apps.gatekeeper.view.fragment.SettingsFragment
import technology.mainthread.apps.gatekeeper.view.fragment.UnlockFragment

@Module
internal abstract class AndroidBindingModule {

    // Activity
    @ContributesAndroidInjector
    internal abstract fun mainActivity(): MainActivity

    @ContributesAndroidInjector
    internal abstract fun authActivity(): AuthActivity

    @ContributesAndroidInjector
    internal abstract fun settingsActivity(): SettingsActivity

    // Fragment
    @ContributesAndroidInjector
    internal abstract fun imagesFragment(): ImagesFragment

    @ContributesAndroidInjector
    internal abstract fun logsFragment(): LogsFragment

    @ContributesAndroidInjector
    internal abstract fun settingsFragment(): SettingsFragment

    @ContributesAndroidInjector
    internal abstract fun unlockFragment(): UnlockFragment

    // Service
    @ContributesAndroidInjector
    internal abstract fun gatekeeperStateService(): GatekeeperStateService

    @ContributesAndroidInjector
    internal abstract fun messagingInstanceIdService(): MessagingInstanceIdService

    @ContributesAndroidInjector
    internal abstract fun messagingService(): MessagingService

    @ContributesAndroidInjector
    internal abstract fun refreshFCMSubscriptionsService(): RefreshFCMSubscriptionsService
}
